cpp/src/utils.cpp

```diff
@@ -74,19 +75,15 @@ bool parse_string(std::string const& value) {
 
 int get_current_numa_node_id() {
     static const int numa_node_id = [] {
-        RAPIDSMPF_EXPECTS(nvmlInit() == NVML_SUCCESS, "failed to initialize NVML");
-        nvmlDevice_t device;
         RAPIDSMPF_EXPECTS(
-            nvmlDeviceGetHandleByIndex(0, &device) == NVML_SUCCESS,
-            "failed to get device handle"
+            numa_available() >= 0, "NUMA is not available", std::runtime_error
         );
-        uint32_t numa_node_id = 0;
+        int cpu = sched_getcpu();
```

Comment:
Ok, I think this works but _only if the CPU affinity has been properly set when launching the process_. Could you document that?